### ISSUE TITLE
 Environment.Exit on error

### ISSUE THREAD

**Issue by JeremiePatron** [christophediericx/ArduinoSketchUploader#12]
Hello,

When i use ArduinoSketchUploader in my C# app , I don't want Environment.Exit on error but just a log or a Exception.
It's possible to add an option for this ?   

```
internal static void LogErrorAndQuit(string errorMessage) 
        {
            logger.Error(errorMessage);
            Environment.Exit(1);
        }
```

**Comment by christophediericx** [comment]
Hi,

Sure, that's a valid use case when using as a library. I have a couple of very busy days ahead, but I will try to make the required changes over the weekend!

Thanks,

**Comment by JeremiePatron** [comment]
Nice, I will wait your changes. 
Thank you.

**Comment by christophediericx** [comment]
Hi @JeremiePatron,

Sorry for the wait! This has now been implemented: an exception in the library will log and throw an `ArduinoUploaderException` rather than causing an `Environment.Exit` (the consuming code should decide how to handle this exception). The version this behavior is published in is 2.4.5 (might take a 10 minutes or so for the package update to become available on nuget.org).

Thanks again for your interest in this library!

Christophe

**Comment by JeremiePatron** [comment]
Thank you.
It's working.
